db/sqlglue.c

```diff
@@ -8194,15 +8207,15 @@ int sqlite3BtreeCursor(
 
     if (pBt->is_temporary) { /* temp table */
         assert(iTable >= 1); /* can never be zero or negative */
-        int pgno = iTable;
+        Pgno pgno = iTable;
         if (forOpen) {
             /*
             ** NOTE: When being called to open a temporary (table) cursor in
             **       response to OP_OpenRead, OP_OpenWrite, or OP_ReopenIdx
             **       opcodes by the VDBE, the temporary table may not have
             **       been created yet.  Attempt to do that now.
             */
-            if (hash_find(pBt->temp_tables, &pgno) == NULL) {
+            if (hash_find(pBt->temp_tables, &iTable) == NULL) {
```

Comment:
pgno?

Reply:
Technically, it's not of the right type (even though both variables should have the same value at this point.